keycloak/openid_client.go

```diff
@@ -1,37 +1,329 @@
+	StandardFlowEnabled          bool                               `json:"standardFlowEnabled"`
+	ImplicitFlowEnabled          bool                               `json:"implicitFlowEnabled"`
+	DirectAccessGrantsEnabled    bool                               `json:"directAccessGrantsEnabled"`
+	ServiceAccountsEnabled       bool                               `json:"serviceAccountsEnabled"`
+	AuthorizationServicesEnabled bool                               `json:"authorizationServicesEnabled"`
+	ValidRedirectUris            []string                           `json:"redirectUris"`
+	WebOrigins                   []string                           `json:"webOrigins"`
+	AuthorizationSettings        *OpenidClientAuthorizationSettings `json:"authorizationSettings,omitempty"`
+}
+
+func (keycloakClient *KeycloakClient) GetClientAuthorizationPolicyByName(realmId, resourceServerId, name string) (*OpenidClientAuthorizationPolicy, error) {
+	policies := []OpenidClientAuthorizationPolicy{}
+	params := map[string]string{"name": name}
+	err := keycloakClient.get(fmt.Sprintf("/realms/%s/clients/%s/authz/resource-server/policy", realmId, resourceServerId), &policies, params)
+	if err != nil {
+		return nil, err
+	}
+	policy := policies[0]
+	policy.RealmId = realmId
+	policy.ResourceServerId = resourceServerId
+	policy.Name = name
+	return &policy, nil
+}
+
+func (keycloakClient *KeycloakClient) GetOpenidClientAuthorizationPermission(realm, resourceServerId, id string) (*OpenidClientAuthorizationPermission, error) {
+	permission := OpenidClientAuthorizationPermission{
+		RealmId:          realm,
+		ResourceServerId: resourceServerId,
+		Id:               id,
+	}
+
+	policies := []OpenidClientAuthorizationPolicy{}
+	resources := []OpenidClientAuthorizationResource{}
+
+	err := keycloakClient.get(fmt.Sprintf("/realms/%s/clients/%s/authz/resource-server/permission/resource/%s", realm, resourceServerId, id), &permission, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = keycloakClient.get(fmt.Sprintf("/realms/%s/clients/%s/authz/resource-server/policy/%s/associatedPolicies", realm, resourceServerId, id), &policies, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = keycloakClient.get(fmt.Sprintf("/realms/%s/clients/%s/authz/resource-server/permission/%s/resources", realm, resourceServerId, id), &resources, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, policy := range policies {
+		permission.Policies = append(permission.Policies, policy.Id)
+	}
+
+	for _, resource := range resources {
+		permission.Resources = append(permission.Resources, resource.Id)
+	}
+
+	return &permission, nil
+}
+
+func (keycloakClient *KeycloakClient) NewOpenidClientAuthorizationPermission(permission *OpenidClientAuthorizationPermission) error {
+	body, _, err := keycloakClient.post(fmt.Sprintf("/realms/%s/clients/%s/authz/resource-server/permission", permission.RealmId, permission.ResourceServerId), permission)
```

Comment:
it looks like this was the only reason the `keycloakClient.post` signature changed.  Could you help me understand why the response body is important here?

Reply:
It's needed to get permission id from response body. Id cannot be extracted from location header as it's done for the rest of resources